File: backend/Api/Models/Supplier.cs
using System.Text.Json.Serialization;

namespace Api.Models
{
    public class Supplier
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? ContactPerson { get; set; }
        public string Phone { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public string? Address { get; set; }
        public string? Category { get; set; }
        public decimal Rating { get; set; } = 0;
        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [JsonIgnore]
        public virtual ICollection<PurchaseOrder> PurchaseOrders { get; set; } = new List<PurchaseOrder>();
    }
}

File: go-service/models/supplier.go
package models

import "time"

// Supplier represents a supplier/vendor
type Supplier struct {
	ID            int        `db:"Id" json:"id"`
	Name          string     `db:"Name" json:"name"`
	ContactPerson *string    `db:"ContactPerson" json:"contactPerson,omitempty"`
	Phone         string     `db:"Phone" json:"phone"`
	Email         string     `db:"Email" json:"email"`
	Address       *string    `db:"Address" json:"address,omitempty"`
	Category      *string    `db:"Category" json:"category,omitempty"`
	Rating        float64    `db:"Rating" json:"rating"`
	IsActive      bool       `db:"IsActive" json:"isActive"`
	CreatedAt     time.Time  `db:"CreatedAt" json:"createdAt"`
	UpdatedAt     *time.Time `db:"UpdatedAt" json:"updatedAt,omitempty"`
}
